Case insensitive search for textures

AlexanderVeselov opened the pull request.

dtarakanov1 committed:
Data generator: enable loading light textures with relative paths

Commit:
Fixed NAN in outputs

AlexanderVeselov committed:
Remove conditional copying of debug dll on Windows

AlexanderVeselov committed:
Change dependencies of utility projects

Change dependencies to fix build errors on Windows because of
different build order

AlexanderVeselov committed:
Format CMakeLists files

Commit:
Updated Render and config_loader classes api. Added new commanl light options.

Commit:
Changed camera indexing

Commit:
Added partial dataset loading support

AlexanderVeselov committed:
Fix Ninja generation error

o Replace CMAKE_RUNTIME_OUTPUT_DIRECTORY to CMAKE_BINARY_DIR
- Remove DEPENDS keyword (it didn't work)

Commit:
Code review

AlexanderVeselov committed:
Fix make install error on Windows

AlexanderVeselov committed:
Fix DataGenerator CMakeLists, add installation for tests

AlexanderVeselov committed:
Fix CMakeLists on Linux

+ Add empty Copy3rdParty project
o Fix BaikalTest and RprTest directories

AlexanderVeselov committed:
Add install commands for RadeonProRender64 and RprSupport64

AlexanderVeselov committed:
Revert cosmetic changes

AlexanderVeselov committed:
Revert more whitespaces

Commit:
Chaned camera set distirbution logic

AlexanderVeselov committed:
Revert more cosmetic changes

AlexanderVeselov committed:
Fix path to release OIIO dll

Commit:
Code review comments. Replaced uint32_t with size_t. Added log to console

Commit:
Code review

Commit:
Deleted changing camera pos

Commit:
Removed checks

Commit:
Changed checks and fixed orthographic mistakes

Commit:
Added template magic

Commit:
Added metadata saving

AlexanderVeselov committed:
Update ImGui & tinyobj to make Baikal be compilable on g++-8 (#183)

o Move tiny_obj_loader to 3rdparty directory

Commit:
Added num_bounce cmd option into BaiaklDataGenerator

AlexanderVeselov committed:
Change zero uv generation

o Do not defer filling texcoords with zeroes, do it immediately: add zero to texcoord vector if texcoord index equals to -1

yozhijk committed:
Merge pull request #191 from AvKhokhlov/metadata_for_data_generator

Metadata for data generator

yozhijk committed:
Merge pull request #180 from dtarakanov1/master

Data generator: enable loading light textures with relative paths

yozhijk committed:
Merge pull request #182 from AvKhokhlov/outputs_nan_fix

Fixed NAN in outputs

Commit:
Merge

AlexanderVeselov committed:
Add case insensitive search of textures to BaikalIO

AlexanderVeselov committed:
Add stdc++fs library only on Unix

AlexanderVeselov committed:
Remove 'Tf' token from MTL parser to save compatibility

AlexanderVeselov committed:
Merge branch 'update_tinyobj_imgui' into case_insensitive_textures

yozhijk committed:
Merge pull request #185 from AvKhokhlov/Improve_data_generator

Improve data generator

yozhijk committed:
Merge pull request #189 from AlexanderVeselov/fix_cmakelists

Another CMakeLists sctipt update

Kvalme commented on BaikalIO/file_utils.cpp:
```diff
@@ -0,0 +1,152 @@
+
+namespace Baikal
+{
+    namespace fs = std::experimental::filesystem;
```
better use filesystem or even remove this shortcut

AlexanderVeselov committed:
Add preprocessor conditional compilation for FindFilenameFromCaseInsensitive

• Mac OSX Xcode compiler library doesn't support filesystem. On the other hand, OSX filesystem is case insensitive, so we don't need upper/lower case conversion as well as on Windows. So we add #ifdef macro to compile FindFilenameFromCaseInsensitive(...) function only on Linux.

AlexanderVeselov committed:
Remove 'fs' shortcut

AlexanderVeselov committed:
Merge branch 'master' into case_insensitive_textures

AlexanderVeselov committed:
Fix Linux release build warning
